### ISSUE TITLE
 Adding new flight mode

### ISSUE THREAD

**Issue by Bggopal** [ArduPilot/apm_planner#1196]
Hi guys,
I am trying to add a new flight mode in Copter-3.5 using the documentation provided in ("https://ardupilot.org/dev/docs/apmcopter-adding-a-new-flight-mode-35.html"), but even after updating every field (by creating a new flight mode similar to stabilize) i couldn't see it in APM planner, i using APM planner v2.0.23 in linux. the most searchers where relating to a file in mission planner "ParameterMetaData.xml" but for me here it is "apm.pdef.xml" . Even though i edited it, it does not work. Was i on the right way, or where did i do mistake, if you any one have good experience, please provide your help here. Thank you.

Git repo for the changes i did "https://github.com/Bggopal/ardupilot/tree/Copter-3.5"

[x] Pixhawk
[x] Copter
![image](https://user-images.githubusercontent.com/59637425/73587264-fb231e80-44df-11ea-980a-662f4d6d8f4b.png)

It would be great of you to help Mr. @muramura

**Comment by muramura** [comment]
@Bggopal san.
I update two files to add a new flight mode.
I can find two files on ZIGZAG.
I add a new flight mode to these two files.

![Screenshot from 2020-02-02 10-56-08](https://user-images.githubusercontent.com/646194/73601802-40f8e900-45ab-11ea-81b3-e0dbe70adde1.png)

**Comment by Bggopal** [comment]
Thank you Mr. @muramura  ,  it worked, but one more small issue is that after trying to implement the created new mode it shows FAILURE: failed cmd 11, in apm planner. I Think it is some part error with firmware programming. What i need to change that. Though i have changed the files for new flight mode in  control_newflightmode.cpp and called the functions crtly.
![Screenshot from 2020-02-03 16-45-12](https://user-images.githubusercontent.com/59637425/73714668-9f52d280-4737-11ea-9af2-3310bfcce9b7.png)

**Comment by muramura** [comment]
@barthess san.
I just added the definition and processing to APM Planner2 only, the flight controller does not recognize that number.
Add definitions and processing to the flight controller as well.
I defined TESTMODE (127), specified POSHOLD for processing, and built the firmware.
I specified TESTMODE from APM PLANNER2 and confirmed that the flight mode became TESTMODE.
I was able to fly TESTMODE at SITL.

![Screenshot from 2020-02-11 16-19-25](https://user-images.githubusercontent.com/646194/74217366-8c697080-4cea-11ea-8987-8c0751fbc1aa.png)

![Screenshot from 2020-02-11 16-18-59](https://user-images.githubusercontent.com/646194/74217330-7e1b5480-4cea-11ea-9bb7-2f76b3d80dfc.png)
![Screenshot from 2020-02-11 16-18-55](https://user-images.githubusercontent.com/646194/74217311-7360bf80-4cea-11ea-8921-65b7cba409ac.png)
